app/src/main/java/org/sopt/cgv/feature/time/component/TheaterSelectionModalBottomSheet.kt

```diff
@@ -0,0 +1,131 @@
+package org.sopt.cgv.feature.time.component
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.width
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.ModalBottomSheet
+import androidx.compose.material3.SheetState
+import androidx.compose.material3.rememberModalBottomSheetState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.mutableIntStateOf
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import kotlinx.collections.immutable.persistentListOf
+import org.sopt.cgv.core.designsystem.theme.White
+
+@ExperimentalMaterial3Api
+@Composable
+fun TheaterSelectionModalBottomSheet(
+    isSheetOpen: MutableState<Boolean>,
+    onDismissRequest: () -> Unit,
+    sheetState: SheetState,
+    selectedTabInModalIndex: MutableState<Int>,
+    selectedRegionInModal: MutableState<String>,
+    selectedTheaters: MutableState<Set<String>>,
+    modifier: Modifier = Modifier
```

Comment:
```suggestion
   isSheetOpen: Boolean,
   selectedTabIndex: Int,
   selectedRegion: String,
   selectedTheaters: Set<String>
   
   onTabSelected: (Int) -> Unit,
   onRegionSelected: (String) -> Unit,
   onTheaterSelected: (String) -> Unit,
   onTheaterDeselected: (String) -> Unit
```
(MutableState만 수정)(이벤트 핸들링 추가)

이렇게 구현하면 MutableState를 직접 전달하는 대신 불변 값을 전달하고 상태 관리 책임을 컴포넌트 외부로 이동시킬 수 있습니다!
그리고 각각의 이벤트에 대한 콜백 함수 추가 상태 변경 로직을 외부로 위임하여
컴포넌트를 UI 표시와 이벤트 전달만 담당할 수 있도록 구현 가능할 거 같아요.